lib/models/room-state.js

```diff
@@ -225,6 +225,52 @@ RoomState.prototype.getUserIdsWithDisplayName = function(displayName) {
 };
 
 /**
+ * Returns true if the given user ID has permission to send a state
+ * event of type `stateEventType` into this room.
+ * @param {string} type The type of state events to test
+ * @param {string} userId The user ID of the user to test permission for
+ * @return {boolean} true if the given user ID should be permitted to send
+ *                        the given type of state event into this room,
+ *                        according to the room's state.
+ */
+RoomState.prototype.maySendStateEvent = function(stateEventType, userId) {
+    if (!this.getMember(userId)) { return false; }
+
+    var power_levels_event = this.getStateEvents('m.room.power_levels', '');
+
+    var power_levels;
+    var events_levels = {};
+
+    var default_user_level = 0;
+    var user_levels = [];
+    var current_user_level = 0;
+
+    var state_default = 0;
```

Comment:
why is this `0`, but `50` at line 262? it is odd that the required power level is `50` if there is an `m.room.power_levels` with no `state_default`, but `0` if there is no `m.room.power_levels` at all

(It is also odd that the spec doesn't clarify this)